main.py

```diff
@@ -0,0 +1,33 @@
+import time
```

Comment:
How about creating a `packages` directory and inside it creating e.g. `xmpp_server` and `spade_agents` packages and moving all the related stuff to their own packages? As of now everything is mixed up and it's getting hard to navigate through all of those files.

Also https://docs.docker.com/compose/extends/#extending-services could be useful, so that we would have only a single reference to each package in a root `docker-compose.yml` and all the references to e.g. `./prosody` or `./prosody_db` would stay intact in a package-scoped `docker-compose.yml`.